### ISSUE TITLE
Implement functions for checking custom action conditions

### ISSUE THREAD

**Issue by brentd313** [uchicago-cs/chiventure#553]
I will be writing the implementations of the fundamental condition-checking functions to be used with custom actions. The skeleton functions can currently be found in "atomic.c" in the custom-actions module of the "custom-actions/intial-headers" branch. I will be testing these functions as I implement them.

**Comment by brentd313** [comment]
I opened a new branch from dev, titled "custom-actions/conditions-effects", on which I have completed the implementation for the fundamental condition-checking functions for custom action execution. The only function that still needs to be implemented is check_in, which checks whether an item is in a container. I will need to investigate how to access and check the list of items associated with a container in order to implement check_in.

**Comment by brentd313** [comment]
The implementation of the condition checking functions can be found in the "custom-actions-cond.c" file in the src/custom-actions/src directory. Additionally, I have created a "custom-actions-cond.h" header and a "custom-actions-common.h" file in the src/custom-actions/include directory.

**Comment by brentd313** [comment]
I wrote unit tests for each of the condition-checking functions implemented in "custom-actions-cond.c". These tests can be found in the file "test_ca_cond.c" in the tests/custom-actions directory of the branch mentioned above.

**Comment by brentd313** [comment]
I modified the existing CMakeLists.txt files and created new ones in src/custom-actions and tests/custom-actions in order to include the condition-checking functions and their tests in the chiventure build.

**Comment by brentd313** [comment]
After addressing the errors raised during the compilation of the condition-checking functions and their tests, the code is compiling error-free and all tests are passing.

**Comment by brentd313** [comment]
The pull request with the fundamental condition-checking functions has been merged into the dev branch, and the custom-actions/conditions branch has been deleted. To review, these functions provide basic ways to check conditions that will determine whether or not a custom action can be executed or not. In this issue, I implemented these functions and wrote unit tests. While these provide a good set of tools for checking conditions of a function, we will continue to update these implementations with improvements or new functions as need for new functionality presents itself. The check_in() function described in "custom-actions-cond.h" provides a good example of what a more advanced condition-checking function might be for future implementation.
